core/commands/commit.py

```diff
@@ -110,22 +111,23 @@ def run(self, edit):
                 initial_text += f.read()
 
         show_commit_diff = savvy_settings.get("show_commit_diff")
+        initial_git_command = [
+            "diff",
+            "--no-color"
+        ]
+
         if show_commit_diff == "stat":
-            initial_text += self.git(
-                "diff",
-                "--stat",
-                "--no-color",
-                "--cached",
-                "HEAD^" if option_amend else None
-            )
-        elif show_commit_diff == "full" or show_commit_diff is True:
-            initial_text += self.git(
-                "diff",
-                "--no-color",
-                "--cached",
-                "HEAD^" if option_amend else None
-            )
+            initial_git_command.append("--stat")
+
+        if not include_unstaged:
+            initial_git_command.append("--cached")
+
+        if option_amend:
+            initial_git_command.append("HEAD^")
+        elif include_unstaged:
+            initial_git_command.append("HEAD")
 
+        initial_text += self.git(*initial_git_command) if show_commit_diff != 'false' else ''
```

Comment:
calling it `initial_git_command` seems counterintuitive as it is the final git_commands...perhaps we could just call it `args`. For the same reason, `initial_text` should be renamed differently, maybe `diff_text`?

Reply:
I agree better call it git_args

Reply:
naming is hard :P